pkg/plugins/diff/service.go

```diff
@@ -136,3 +133,41 @@ func (s *Service) appendClosingFunction(diffType string, f func()) {
 		s.closeFunctions[diffType] = f
 	}
 }
+
+// NewService is used to initialize a new Differ service. The returned function is a closing function for the service.
+func NewService(diffProps map[string]config.DiffProps, pluginProps map[string]config.PluginProps, pluginsPath string) (*Service, func()) {
+	service := &Service{
+		pluginHandler:  internal.NewManager[Differ](),
+		closeFunctions: make(map[string]func()),
+	}
+	registerPlugins(service, diffProps, pluginProps, pluginsPath)
+	return service, service.Close
+}
+
+func registerPlugins(service *Service, diffProps map[string]config.DiffProps, pluginProps map[string]config.PluginProps, pluginsPath string) {
+	if !strings.HasSuffix(pluginsPath, "/") {
+		pluginsPath += "/"
+	}
+	for n, p := range diffProps {
+		pluginName := p.PluginName
+		// If the requested plugin wasn't configured with a path, it will be defined under the default location
+		pluginPath := pluginsPath + pluginName
+		pluginVersion := 1 // default version
+		if props, ok := pluginProps[pluginName]; ok {
+			pluginPath = props.Path
+			if props.Version != nil {
+				pluginVersion = *props.Version
+			}
+		}
+
+		if strings.ToLower(n) == "delta" {
+			pid := plugins.PluginIdentity{ProtocolVersion: uint(pluginVersion), ExecutableLocation: pluginPath}
+			pa := plugins.PluginHandshake{}
+			RegisterDeltaLakeDiffPlugin(service, pid, pa)
+		} else {
+			logging.Default().Warnf("failed to register a plugin for an unknown diff type: '%s'", n)
+			return
```

Comment:
We normally fail when the config isn't right. Shouldn't we stop the app in this condition? For example:

```
  diff:
      <valid_config>
  bad-name:
     ....
```
Will yield an app with a single working plugin. But switch the plugins:
```
  bad-name:
     ....
  diff:
      <valid_config>
```
And you end up with a running lakeFS with 0 plugins.

Reply:
I see.
I think that we shouldn't stop the application just because the user tried to register a plugin that doesn't exist.
I'll remove the `return` and raise the log level to `error`.

Reply:
We normally do on wrong configuration file. Up to you, we can also change it later on.